### ISSUE TITLE
Encountered two children with the same key

### ISSUE THREAD

**Issue by sunny-mittal** [LegendApp/legend-state#14]
Hello,
I've noticed when using Legend that if I render (in React Native) a Text element with two pieces of state from an observable, I get a warning that two children with the same key were found:
```
<Text>{user.firstname} {user.lastname}</Text>
```
They key that it warns about is an empty string and there's only one text element, but I think it has to do with how React Native virtualizes text components (so the two parts of the name become two virtual text elements). Does Legend do something involving keys when rendering? If I remove the lastname bit from the above or if I replace it with any two pieces of static text, there's no warning. There's also no warning if I call `get` on each of the name pieces so it seems to have something to do with the auto-rendering. It'd be nice if I didn't have to sometimes call `get` and sometimes not, as this'll confuse my team a lot :)

**Comment by sunny-mittal** [comment]
Another option would be to destructure, which also "unwraps" the proxy and works:
```
const { firstname, lastname } = user.get()
```
This might be the better solution anyway as it's more readable. Still curious about the main issue though when you have time. Thanks!

**Comment by jmeistrich** [comment]
Oh interesting! I will fix this in the next release.

What's happening is that rendering an observable directly creates a memoized self-updating text component with the key "". I hadn't even thought of having two of those within the same component, so that was a mistake. The fix will be simple, so I will get that in soon.

If you don't see any visible problems I think it should be ok to do as you were and ignore the warning. The underlying bug will be fixed soon so the warning will go away.

**Comment by sunny-mittal** [comment]
Awesome, thank you so much! And thank you for being so fast to respond! Awesome project, by the way. The use of proxies is just brilliant.

**Comment by jmeistrich** [comment]
Thanks! I hope it's working well for you other than this bug :). I'll just reopen this and close it when the fix is released.

**Comment by chakravarthi-bb** [comment]
I think the better way is to access raw values when rendering Text

```
<Text>{user.firstName.get()} {user.lastName.get()}</Text>
```

@jmeistrich I don't think any fix is necessary but rather the mental model on How you render observable in React.

**Comment by jmeistrich** [comment]
It's fixed already :). I just set key to `null` instead of `''`. https://github.com/LegendApp/legend-state/commit/69e1e563046774a0c6af2eec70f0c59b8df5b887

Which way is better depends on what you want, and there's tradeoffs. Rendering the observables directly as @sunny-mittal did would make the component not need to render, but it also implicitly creates auto-subscribing sub-components. If the parent is already subscribing to those properties anyway, sub-components that also subscribe is wasteful. But if isolating renders to the sub-components keeps the parent component from rendering as much then it might be better.

I think we should write up more about that tradeoff in the docs.

**Comment by jmeistrich** [comment]
This should be fixed in 0.19.0. But please let me know if you still encounter it. Thanks for the report!
